Interfaces/GCClassic/main.F90

```diff
@@ -1658,6 +1659,42 @@ PROGRAM GEOS_Chem
              WRITE( 6, '(a)' ) REPEAT( '#', 79 )
           ENDIF
 
+          ! Allocate temperature difference arrays (crb, 14/02/24)
+          If ( Input_Opt%RRTMG_FDH ) THEN
+             Allocate(DT_3D(State_Grid%NX,State_Grid%NY,State_Grid%NZ),Stat=RC)
+             IF ( RC /= 0 ) Call Error_Stop( 'Error allocating DT_3D', ThisLoc )
+
+             ! If using seasonally evolving FDH, need to grab the internal
+             ! state temperature adjustment array
+             IF (Input_Opt%RRTMG_SEFDH) THEN
+                ! DT_3D will be updated by the first call to RRTMG; also need to
+                ! store the "current" value
+                Allocate(DT_3D_UPDATE(State_Grid%NX,State_Grid%NY,State_Grid%NZ),Stat=RC)
+                IF ( RC /= 0 ) Call Error_Stop( 'Error allocating DT_3D_UPDATE', ThisLoc )
+                ! Store the adjustment as previously projected to this time point
+                DT_3D(:,:,:) = State_Chm%TStrat_Adj(:,:,:)
+                ! This will just hold the end-of-step value
+                DT_3D_UPDATE(:,:,:) = 0
+             ELSE
+                DT_3D(:,:,:) = 0
+             END IF
+
+             Allocate(HR_3D(State_Grid%NX,State_Grid%NY,State_Grid%NZ),Stat=RC)
+             IF ( RC /= 0 ) Call Error_Stop( 'Error allocating HR_3D', ThisLoc )
+             HR_3D(:,:,:) = 0
+
+             ! Read in dynamical heating rates if necessary
+             IF (Input_Opt%Read_Dyn_Heating) THEN
+                HR_3D(:,:,:) = State_Met%DynHeating(:,:,:)
+             ENDIF
+          ELSE
+             ! Safer
+             Allocate(DT_3D(0,0,0),Stat=RC)
+             IF ( RC /= 0 ) Call Error_Stop( 'Error allocating DT_3D', ThisLoc )
+             Allocate(HR_3D(0,0,0),Stat=RC)
+             IF ( RC /= 0 ) Call Error_Stop( 'Error allocating HR_3D', ThisLoc ) 
+          ENDIF
+
```

Comment:
@cbarker211 Could this be placed into a subroutine after the `CONTAINS` statement in `main.F90` and just called from here?  That would reduce clutter in an already-long `main.F90` program.